k2/csrc/ragged_ops.h

```diff
@@ -852,6 +902,56 @@ template <typename T>
 Ragged<T> Stack(int32_t axis, int32_t num_srcs, Ragged<T> *src,
                 Array1<uint32_t> *merge_map = nullptr);
 
+/*
+  Unstack a Ragged tensor to a list of Ragged, tensors all the output Ragged
+  tensors have one less axis. Similar to TF's Unstack (or unbind in Pytorch).
+
+    @param [in] src  The ragged tensor to be unstacked.
+    @param [in] axis  The axis to be removed, all the elements of this axis will
+                      be rearranged into output Raggeds.
+    @param [out] out  The container where the output ragged tensors would write
+                      to. MUST NOT be a nullptr, will be reallocated.
+    @param [out] split_map  If not nullptr will store the indexes mapping from
+                            the splited Raggeds to the original Ragged.
+                            Will be reallocated.
+
+  Caution: If `src.NumAxes() == 2`, the output shapes will only have one
+           dimension, to make it a ragged tensor, we will add a TrivialShape on
+           each of the output tensors.
+
+  Note: The output ragged tensors will contain empty lists on axis `axis`,
+        you can remove them by RemoveEmptyLists if needed.
+
+  The number of output ragged tensors is decided by the max size of sublists on
+  axis `axis`.
+
+  A small example of unstacking a 3 axes Ragged:
+
+    src: [ [ [ 1 2 ] [ 3 ] ] [ [ 4 ] ] ]
+    unstack on axis 0:
+    one sublist on axis 0, its size is 2, will produce 2 ragged tensors.
```

Comment:
I don't understand this, i.e. the behavior of Unstack with different numbers of axes.